pkg/git/WebhookSecretValidator.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020 Devtron Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package git
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"github.com/devtron-labs/devtron/pkg/pipeline"
+	"go.uber.org/zap"
+	"net/http"
+	"strings"
+)
+
+type WebhookSecretValidator interface {
+	ValidateSecret(r *http.Request, secretInUrl string, requestBodyBytes []byte, gitHost *pipeline.GitHostRequest) bool
+}
+
+type WebhookSecretValidatorImpl struct {
+	logger               *zap.SugaredLogger
+}
+
+func NewWebhookSecretValidatorImpl(Logger *zap.SugaredLogger) *WebhookSecretValidatorImpl {
+	return &WebhookSecretValidatorImpl{
+		logger:               Logger,
+	}
+}
+
+const (
+	SECRET_VALIDATOR_SHA1 string = "SHA-1"
+	SECRET_VALIDATOR_URL_APPEND string = "URL_APPEND"
+	SECRET_VALIDATOR_PLAIN_TEXT string = "PLAIN_TEXT"
+)
+
+
+func (impl *WebhookSecretValidatorImpl) ValidateSecret(r *http.Request, secretInUrl string, requestBodyBytes []byte, gitHost *pipeline.GitHostRequest) bool {
```

Comment:
add some documentation around this block

Reply:
Done